verifier/src/x509/name.rs

```diff
@@ -0,0 +1,391 @@
+// Copyright (c) 2023 The MobileCoin Foundation
+
+//! X509 distinguished name as defined in sections
```

Comment:
I'm up for suggestions on what else to bring from the RFC to have on hand. I wanted to avoid too much duplication, but  also understand it's nice to have some things closer at hand.